feature/HomeScreen

kiasha10 opened the pull request:
Feature/Merge] - feature/HomeScreen

[Merge] feature/homeScreen → Dev
 [Feature] - homeScreen

 ## Description
Added TableViews
Added UIImageView


## Checklist
Please review and check the following before submitting this pull request:
- [ ] The code follows the project's coding standards.
- [ ] 100% Unit test coverage for View Models

kiasha10 committed:
TableView and UIImageView for feature/HomeScreen. Created an UIImageViewExtension folder to display images. Added some UI changes to storyboard

kiasha10 committed:
Merge branch 'dev' into feature/homeScreen

kiasha10 committed:
Revert "Merge branch 'dev' into feature/homeScreen"

This reverts commit 992c4c368e168b08e4b45fdc67b5839c8cb18f23, reversing
changes made to d2ce870e8fbc8166b652714ada7a0f31c53aa642.

kieran-woodrow commented on BlitzBoard/Features/HomeScreenTableCell/HomeScreenTableViewCell.swift:
```diff
@@ -11,20 +11,21 @@ class HomeScreenTableViewCell: UITableViewCell {
-    @IBOutlet private weak var teamNameLabel: UILabel!
-    @IBOutlet private weak var winsLabel: UILabel!
+    @IBOutlet weak var teamLogo: UIImageView!
+    @IBOutlet weak var teamName: UILabel!
```
private please

kieran-woodrow replied:
please don't resolve this until you have made changes

kieran-woodrow commented on BlitzBoard/Features/HomeScreenTableCell/HomeScreenTableViewCell.swift:
```diff
@@ -11,20 +11,21 @@ class HomeScreenTableViewCell: UITableViewCell {
-        drawsLabel.text = "Draws: \(teamStanding.overallLeagueDraw)"
-        lossesLabel.text = "Losses: \(teamStanding.overallLeagueLoss)"
+        let shorthandTeamName = convertToShorthand(teamName: teamStanding.teamName)
+        teamName.text = shorthandTeamName
```
teamName.text = convertToShorthand(teamName: teamStanding.teamName)
  Just to remove a line of code

kieran-woodrow replied:
same here

kieran-woodrow commented on BlitzBoard/Features/HomeScreenViewController.swift:
```diff
@@ -31,12 +30,46 @@ class HomeScreenViewController: UIViewController {
         tableView.dataSource = self
         tableView.delegate = self
+        
+        tableView.layer.borderWidth = 5.0
```
this can be done in the storyboard. Please google how to add it to the keyPath/attributes inspector

kiasha10 replied:
Will fix UI at a later stage

kieran-woodrow commented on BlitzBoard/Features/HomeScreenViewController.swift:
```diff
@@ -31,12 +30,46 @@ class HomeScreenViewController: UIViewController {
+        let headerView = UIView(frame: CGRect(x: 0, y: 0, width: tableView.frame.width, height: 50))
+        headerView.backgroundColor = .darkGray
+        
+        let positionLabel = UILabel(frame: CGRect(x: 10, y: 0, width: 50, height: 50))
```
This is not wrong. But Why are you declaratively building UI and not doing it in the storyboard?

kiasha10 replied:
will get back to UI

kieran-woodrow commented on BlitzBoard/Features/HomeScreenViewModel.swift:
```diff
@@ -33,15 +33,15 @@ class HomeScreenViewModel {
+            case .success(let teamStandings):
+                self?.leagueTables = teamStandings
                 self?.delegate?.reloadView()
                 print("Here are the stats: \(String(describing: self?.leagueTables))") // #Warning("Will remove print statement in the future ")
```
please add a proper warning without it being in comments
#Warning("Will remove print statement in the future ")
print("Here are the stats: \(String(describing: self?.leagueTables))")

simekanidev replied:
```suggestion
                print("Here are the stats: \(String(describing: self?.leagueTables))")
                 #warning("Will remove print statement in the future ")
```

kieran-woodrow commented on BlitzBoard/Features/SettingsScreenViewController.swift:
```diff
@@ -7,3 +7,11 @@
 
 import UIKit
 
+class SettingsScreenViewController: UIViewController {
```
this file can be removed if not being used

kiasha10 replied:
This will be used

kieran-woodrow replied:
then pleae add a warning. This is in the coding standards we have given you.

kieran-woodrow commented on BlitzBoard/SceneDelegate.swift:
```diff
@@ -12,26 +12,40 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
-        
+        // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
```
can clean up this file and remove the default comments and extra new lines to remove swift lint warnings

kieran-woodrow replied:
here

kieran-woodrow commented on BlitzBoard/UIImageViewExtension.swift:
```diff
@@ -0,0 +1,24 @@
+//  Created by Kiasha Rangasamy on 2024/05/16.
+//
+
+import Foundation
```
can remove foundation

kieran-woodrow replied:
here

kiasha10 committed:
Update BlitzBoard/Features/HomeScreenViewModel.swift

Co-authored-by: Simekani Mabambi <100220689+simekanidvt@users.noreply.github.com>

kiasha10 committed:
PR - reviewed changes

kieran-woodrow commented on BlitzBoard/Features/HomeScreenViewModel.swift:
```diff
@@ -43,8 +43,8 @@ class HomeScreenViewModel {
             case .success(let teamStandings):
                 self?.leagueTables = teamStandings
                 self?.delegate?.reloadView()
+                // #Warning("Will remove print statement in the future ")
```
don't put the //. it mkes it a comment and not a warning

kiasha10 replied:
When I remove the // I get an error saying "No macro named 'Warning'"

kieran-woodrow replied:
small w

kiasha10 replied:
I removed the print statement, just cleaning up code

kiasha10 committed:
HomeScreen UI changes

kieran-woodrow commented on BlitzBoard.xcodeproj/project.pbxproj:
```diff
@@ -10,6 +10,8 @@
 		450CA0AE2BF5E9E0004FF11B /* SDWebImage in Frameworks */ = {isa = PBXBuildFile; productRef = 450CA0AD2BF5E9E0004FF11B /* SDWebImage */; };
 		450CA0B02BF5E9E0004FF11B /* SDWebImageMapKit in Frameworks */ = {isa = PBXBuildFile; productRef = 450CA0AF2BF5E9E0004FF11B /* SDWebImageMapKit */; };
```
Why are there 41 files touched?
Please reas through the standards documents we have put together and the many git lessons we did. We have said quite a few times that PR's should be small (between 1-15) files. Future PR's with more that 10-15 files will be denied on the spot..

kiasha10 replied:
I apologise for the large file, it requested that I fix up my UI in my HomeScreen feature in this PR before merging. I ran into a few issues during my development process which required me to make adjustments in other files. I won't create a large file again.

kieran-woodrow commented on BlitzBoard/Features/HomeScreenTableCell/HomeScreenTableViewCell.swift:
```diff
@@ -25,7 +25,7 @@ class HomeScreenTableViewCell: UITableViewCell {
         drawsLabel.text = "\(teamStanding.overallLeagueDraw)"
         lossesLabel.text = "\(teamStanding.overallLeagueLoss)"
-
+        
```
unnecessary new lines

kiasha10 committed:
PR- removal of unnecessary lines and and indentation

kiasha10 committed:
PR - Added a warning to Settings file

kieran-woodrow commented on BlitzBoard/Features/SettingsScreenViewController.swift:
```diff
@@ -3,7 +3,15 @@
 //  Created by Kiasha Rangasamy on 2024/05/06.
-//
 
+// #warning Settings File will be used in the future
```
#warning Settings File will be used in the future
Don't put a warning in a comment. that defeats the purpose.
Please research how to use warnings in iOS

simekanidev reviewed (commented):
@kiasha10 Is there a reason you are deleting all the files that run your CI/CD pipeline?

kiasha10 committed:
Reapply "Merge branch 'dev' into feature/homeScreen"

This reverts commit 559c30b18dee81a6f781848183fdb5b944a82f50.

# Conflicts:
#	BlitzBoard.xcodeproj/project.pbxproj
#	BlitzBoard/SceneDelegate.swift

kiasha10 committed:
LoginFix on HomeScreen

kiasha10 committed:
Added Test Folder

kiasha10 committed:
Added mock BlitzBoardTests Bundle and fixed indentation on code

kiasha10 committed:
PR - Fixed warning message
